File: hashmap_caching1.php
<?php
// PHPで開発しましょう
function existsWithinList($listL,$dataY){
    $hashMap = array();

    for($i = 0; $i < count($listL);$i++){
        $hashMap[strval($listL[$i])] = $listL[$i];
    }
    return is_null($hashMap[$dataY]) ? false : true;
}

function printBool($bool){
    echo $bool === true ? "True" : "False" .PHP_EOL ;
}

$sampleList = [3,10,23,3,4,50,2,3,4,18,6,1,-2];
echo printBool(existsWithinList($sampleList,23)) .PHP_EOL;
echo printBool(existsWithinList($sampleList,-2)) .PHP_EOL;
echo printBool(existsWithinList($sampleList,100)) .PHP_EOL;

File: hashmap_caching1.py
def existsWithinList(listL,dataY):
    hash_map = {}
    for i in range(int(len(listL))):
        hash_map[listL[i]] = listL[i]
    return False if hash_map.get(dataY) is None else True

sampleList = [3,10,23,3,4,50,2,3,4,18,6,1,-2]
print(existsWithinList(sampleList,23))
print(existsWithinList(sampleList,-2))
print(existsWithinList(sampleList,100))